crates/sui-adapter/src/adapter.rs

```diff
@@ -425,15 +425,16 @@ fn init_modules<
             args.push(bcs_char_type_value);
         }
         args.push(ctx.to_vec());
-        let has_ctx_arg = true;
+        // init must have a mutable
+        let ctx_is_mut = Some(true);
```

Comment:
If we want to allow immutable txn context for `init`, we would have to change this to check the signature of the function for the information. 

That being said, I'm not sure what good an `init` function does without being able to create new objects

Reply:
I think you're right that just about any useful init will have `&mut TxContext` (though I could imagine e.g., emitting an event that contains the epoch where a module was published, which wouldn't need `&mut`).

However, I'd prefer to have the same rules everywhere just so we don't have to explain to programmers that `init` is a special case, even if it makes the code here a bit more complicated. I think it's simpler to explain that you can use both `&TxContext` and `&mut TxContext` everywhere.

Reply:
Fair enough, reverted